editoast/src/modelsv2/connection_pool/db_connection_pool.rs

```diff
@@ -0,0 +1,85 @@
+use diesel_async::pooled_connection::deadpool::Object;
+use diesel_async::pooled_connection::deadpool::Pool;
+#[cfg(test)]
+use diesel_async::pooled_connection::AsyncDieselConnectionManager;
+#[cfg(test)]
+use diesel_async::AsyncConnection;
+use diesel_async::AsyncPgConnection;
+use std::sync::Arc;
+#[cfg(test)]
+use tokio::sync::OwnedRwLockWriteGuard;
+#[cfg(test)]
+use tokio::sync::RwLock;
+#[cfg(test)]
+use url::Url;
+
+use super::DbConnectionError;
+
+#[derive(Clone)]
+pub struct DbConnectionPoolV2 {
+    pub pool: Arc<Pool<AsyncPgConnection>>,
+    #[cfg(test)]
+    pub test_connection: Option<Arc<RwLock<Object<AsyncPgConnection>>>>,
+}
+
+impl DbConnectionPoolV2 {
+    pub fn new(pool: Pool<AsyncPgConnection>) -> Self {
+        Self {
+            pool: Arc::new(pool),
+            #[cfg(all(test))]
+            test_connection: None,
+        }
+    }
+
+    #[cfg(test)]
+    pub async fn new_for_tests(
```

Comment:
Why do we need a `new` and a `new_for_tests`? Why not only a `new` feature-gated? I see that you need `async` and `Result` additionally, but maybe `Result` can be avoided (unwrapping is probably ok for tests). And for the `async` part, I believe we should be able to run the nested functions with `Handle::current().block_on(whatever_asyn_function())`.